rusk-prover/src/lib.rs

```diff
@@ -13,10 +13,30 @@ pub use crate::prover::LocalProver;
 
 pub use errors::ProverError;
 
+use dusk_wallet_core::UnprovenTransaction;
+
 pub type ProverResult = Result<Vec<u8>, ProverError>;
 
 pub trait Prover {
-    fn prove_execute(&self, utx_bytes: &[u8]) -> ProverResult;
+    fn prove_execute(&self, utx_bytes: &[u8]) -> ProverResult {
```

Comment:
Any specific reason to split this into multiple trait fn instead of handle the `match` inside the implementation?

There are use cases (aws lambda prover), where prover implementation have no idea of which circuit it want to prove (it acts as a proxy).

Reply:
No specific reason tbh, I simply thought that being able to directly access the prover might be nice down-stream. If it's not, I'll remove it

Reply:
It's nice, true. 
However I'm still not sure what would be the best approach (hence I've only commented)

I double checked the code for aws prover and it already parse the number of inputs.
So, do what you think is the best :)